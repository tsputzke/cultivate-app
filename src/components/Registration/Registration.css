* {
  box-sizing: border-box;
}

.registration {
  margin: auto;
  padding-top: 200px;
  line-height: 1.5;
  width: 80%;
}

#registration-section {
  padding: 20px;
  text-align: center;
  margin: auto;
  border: 1px solid black;
  width: 300px;
  height: 250px;
}

#new-user-button {
  border: 3px solid red;
}