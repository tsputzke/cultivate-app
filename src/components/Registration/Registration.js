import React, { Component } from 'react';
import { Link } from 'react-router-dom'
import AuthApiService from '../../services/auth-api-service'

export default class Registration extends Component {
  state = { 
    user_name: '',
    password: '',
    userAdded: false,
    error: null,
  };

  handleNewUser = e => {
    e.preventDefault();
    const { user_name, password, confirm_password } = e.target

    // Confirm that passwords match
    if (password.value !== confirm_password.value) {
      alert('passwords must match')
      window.location ='/registration'
    }

    this.setState({ error: null })
      AuthApiService.postUser({
        user_name: user_name.value,
        password: password.value,
      })
        .then(user => {
          user_name.value = ''
          password.value = ''
          this.props.history.push(`/`)
          // window.location ='/show-user'
        })  
        .catch(res => {
          this.setState({ error: res.error })
        })
  }

  render() {
    const { error } = this.state
    return (
      <div className='registration'>
        <form className="registration-form" onSubmit={this.handleNewUser}>
          <div role='alert'>
            {error && <p className='error'>{error}</p>}
          </div>
          <fieldset>
            <legend>Create Account: </legend>
            <div>
              <label htmlFor="user_name">Username: </label>
              <input type="text" name="user_name" />
            </div>
            <div>
              <label htmlFor="password">Password: </label>
              <input type="password" name="password" />
            </div>
            <div>
              <label htmlFor="confirm_password">Confirm Password: </label>
              <input type="password" name="confirm_password" />
            </div>

            <button className="registration-button" type="submit">Submit</button>

            <p>Password must be at least 8 characters long. It must contain at least one number and special character</p>

            <p>Already a user? <span><Link to='/'>login here</Link></span></p>
          </fieldset>
        </form>
      </div>
    );
  }
}
