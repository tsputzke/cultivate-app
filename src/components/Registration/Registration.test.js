import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Registration from './Registration'

configure({adapter: new Adapter()});  

describe(`Registration component`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <Registration />
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders Registration by default', () => {
    const wrapper = shallow(<Registration />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})