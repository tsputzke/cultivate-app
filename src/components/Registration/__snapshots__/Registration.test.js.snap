// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration component renders Registration by default 1`] = `
<div
  className="registration"
>
  <form
    className="registration-form"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend
        className="strong registration-legend"
      >
        Create Account: 
      </legend>
      <table
        className="registration-table"
      >
        <tbody>
          <tr>
            <td>
              <label
                htmlFor="user_name"
              >
                Username:
              </label>
            </td>
            <td>
              <input
                maxLength="20"
                name="user_name"
                type="text"
              />
            </td>
          </tr>
          <tr>
            <td>
              <label
                htmlFor="password"
              >
                Password:
              </label>
            </td>
            <td>
              <input
                maxLength="20"
                name="password"
                type="password"
              />
            </td>
          </tr>
          <tr>
            <td>
              <label
                htmlFor="confirm_password"
              >
                Confirm Password:
              </label>
            </td>
            <td>
              <input
                name="confirm_password"
                type="password"
              />
            </td>
          </tr>
        </tbody>
      </table>
      <button
        className="registration-button"
        type="submit"
      >
        Submit
      </button>
      <p>
        Already a user? 
        <span>
          <Link
            to="/"
          >
            login here
          </Link>
        </span>
      </p>
      <hr />
      <div
        className="password-rules"
      >
        <strong>
          Password must contain: 
        </strong>
        <ul>
          <li>
            8+ characters.
          </li>
          <li>
            At least 1 number.
          </li>
          <li>
            At least 1 special character.
          </li>
        </ul>
      </div>
    </fieldset>
    <div
      role="alert"
    />
  </form>
</div>
`;
